include/codec.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API header file for Audio Codec
+ *
+ * This file contains the Audio Codec APIs
+ */
+
+#ifndef __CODEC_H__
+#define __CODEC_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * PCM audio sample rates
+ */
+typedef enum {
+	CODEC_PCM_RATE_8K	= 8000,
+	CODEC_PCM_RATE_16K	= 16000,
+	CODEC_PCM_RATE_24K	= 24000,
+	CODEC_PCM_RATE_32K	= 32000,
+	CODEC_PCM_RATE_44P1K	= 44100,
+	CODEC_PCM_RATE_48K	= 48000,
+	CODEC_PCM_RATE_96K	= 96000,
+	CODEC_PCM_RATE_192K	= 192000,
+} codec_pcm_rate_t;
+
+/**
+ * PCM audio sample bit widths
+ */
+typedef enum {
+	CODEC_PCM_WIDTH_16_BITS	= 16,
+	CODEC_PCM_WIDTH_20_BITS	= 20,
+	CODEC_PCM_WIDTH_24_BITS	= 24,
+	CODEC_PCM_WIDTH_32_BITS	= 32,
+} codec_pcm_width_t;
+
+/**
+ * Digital Audio Interface (DAI) type
+ */
+typedef enum {
+	CODEC_DAI_TYPE_I2S,
+	CODEC_DAI_TYPE_INVALID,
+} codec_dai_type_t;
+
+/**
+ * Digital Audio Interface (DAI) format
+ */
+typedef enum {
+	CODEC_DAI_MODE_I2S,	/* standard I2S mode */
+	CODEC_DAI_MODE_DSP,	/* DSP mode */
+	CODEC_DAI_MODE_RJF,	/* Right justified mode */
+	CODEC_DAI_MODE_LJF,	/* Left justified mode */
+	CODEC_DAI_MODE_TDM,	/* Multi-channel TDM mode */
+	CODEC_DAI_MODE_INVALID,
+} codec_dai_mode_t;
+
+/**
+ * Codec configuration parameters
+ */
+struct codec_cfg {
+	u32_t			mclk_freq;	/* MCLK input frequency in Hz */
+	codec_pcm_rate_t	pcm_rate;	/* PCM sample rate in Hz */
+	codec_pcm_width_t	pcm_width;	/* PCM sample size in bits */
+	codec_dai_type_t	dai_type;	/* Digital interface type */
+	codec_dai_mode_t	dai_mode;	/* Digital interface mode */
+	u32_t			channels;	/* Number of audio channels */
+	bool 			dai_master;	/* Is codec the DAI master? */
+};
+
+/**
+ * Codec output volume control range
+ * Volume range is from 0dB to -78dB in steps of 0.5dB
+ * Example"
+ * vol = -41 is -20.5dB
+ * vol = -156 is -78dB
+ */
+#define CODEC_OUTPUT_VOLUME_MAX		0
+#define CODEC_OUTPUT_VOLUME_MIN		(-78 * 2)
+
+/**
+ * @brief Configure the audio codec
+ *
+ * Configure the audio codec device according to the configuration
+ * parameters provided as input
+ *
```

Comment:
you'll need to document each parameters here.

What is dev, an i2s device or an audio codec instance?

Reply:
@tbursztyka I'll document each parameter

Reply:
Parameters documented.